wadas/ui/mainwindow.py

```diff
@@ -362,6 +376,17 @@ def url_input_dialog(self):
             logger.warning("URL insertion aborted.")
             return ""
 
+    def custom_species_dialog(self):
+        """Method to run a dialog to return selected custom species to classify."""
+
+        species_dialog = DialogSelectAnimalSpecies()
+        if species_dialog.exec_():
```

Comment:
:=

Reply:
not sure how to create object and call its method on assigment.

Reply:
if (species_dialog := DialogSelectAnimalSpecies()).exec_():